app/scripts/models/account.js

```diff
@@ -90,18 +90,23 @@ define([
       // upgrade the credentials with an accessToken
       promise = promise.then(function () {
         if (self._needsAccessToken()) {
-          return self.createOAuthToken('profile:write')
-            .then(function (accessToken) {
-              self.set('accessToken', accessToken.get('token'));
-            }, function () {
-              // Ignore errors; we'll just fetch again when needed
-            });
+          return self._fetchProfileOAuthToken();
         }
       });
 
       return promise;
     },
 
+    _fetchProfileOAuthToken: function () {
+      var self = this;
+      return self.createOAuthToken('profile:write')
```

Comment:
nit: make constant? It was not before but if we are refactoring this a bit, might be a good idea.